miranda/convert/_data_corrections.py

```diff
@@ -67,6 +69,39 @@ def _iter_vars_key(ds, meta, key, project):
         yield vv, val
 
 
+def correct_time_entries(
+    d: xr.Dataset,
+    split: str = "_",
+    location: int = -1,
+    time_field: str = "Time",
+    freq="D",
+) -> xr.Dataset:
+    filename = d.encoding["source"]
+    date = date_parser(Path(filename).name.split(split)[location])
+    entries = len(d[time_field])
+    date_range = pd.date_range(date, periods=entries, freq=freq)
+
+    d[time_field] = xr.DataArray(
+        time_field,
+        date_range,
+        attrs=dict(units=f"days since {date}", calendar="standard"),
```

Comment:
If the coordinate has a datetime dtype, the `units` and  `calendar` attributes _should not_ be set, they are implicit in the dtype.